django/utils/timesince.py

```diff
@@ -88,6 +88,9 @@ def timesince(d, now=None, reversed=False, time_strings=None, depth=2):
         if count == 0:
             break
         result.append(avoid_wrapping(time_strings[name] % {"num": count}))
+        # prevent errors in long intervals
+        if name == "month" and count > 2:
+            seconds += 10 * 60 * 60
```

Comment:
I think we need more tests to show this is correct for all values of count…  It's not clear to me why a single increment of 10 hours would be equally value if count is 2  or if it's 11.  

One of the initial ideas on the ticket was to make the `seconds` divisor relative to a 30.4 days, based on a a first pass at `count`. That may be too complex. But it would sum so the added `.4` would be greater as `count` increased. Don't we need something like that? 🤔

I'd suggest adding tests for just before and after each month increment to make sure we got the right result. e.g. Jan 1 to Mar 1 is 2 months, ..., Jan 1 to Dec 29 is 11 months 4 weeks, etc.. (These could be in a single case using `subTest()`.)

Reply:
Hi @carltongibson, 

Thanks for reviewing my patch. This is my first contribution

The reason for 10 hours was that 10 hours are 0.4167 days, and the average month duration is 365/12 = 30.4167...so the average month is exactly 30 days and 10 hours long.

The main issue here is that there is a `month` constant but months do not have a constant value... I just did a quick test and for the delta from Jan 1 to Mar 1 the result is not 2 months, but 1 month and 4 weeks (which btw, could be considered at least technically correct).

I am now working on a fix that produces the most humanly correct result possible, but to do so I would need to modify some of the pre-exisisting tests as well. There are some tests in `humanize_tests.tests.HumanizeTests` and `utils_tests.test_timesince.TimesinceTests` that needs to be updated. 

For example, there is one test where:
- `d` is 2012/3/9 
- `now` is `d` plus 30 days, that is 2012/4/8

which is not yet one month but the test expects the result to be one month. The correct result would be 4 weeks and 2 days. I was thinking of fixing this by changing the delta from 30 to 31 days for this test, and implement similar fixes in other tests that may fail for the same reason.

Should I modify these tests?